File: frontend/src/services/doctorService.js
import api from "./api";

const URL = "api/doctors";

const doctorService = {
  getAllDoctors() {
    return api.get(URL);
  },

  getDoctorById(id) {
    return api.get(`${URL}/${id}`);
  },

  createDoctor(doctor) {
    return api.post(URL, doctor);
  },

  updateDoctor(id, doctor) {
    return api.put(`${URL}/${id}`, doctor);
  },

  deleteDoctor(id) {
    return api.delete(`${URL}/${id}`);
  }
};

export default doctorService;

File: backend/src/main/java/com/healthsphere/backend/service/DoctorService.java
package com.healthsphere.backend.service;

import com.healthsphere.backend.dto.DoctorRequest;
import com.healthsphere.backend.dto.DoctorResponse;

import java.util.List;

public interface DoctorService {

    DoctorResponse createDoctor(DoctorRequest request);

    List<DoctorResponse> getAllDoctors();

    DoctorResponse getDoctorById(Long id);

    DoctorResponse updateDoctor(Long id, DoctorRequest request);

    void deleteDoctor(Long id);
}